### ISSUE TITLE
Getting different bounds compare to leaflet using CrsSimple()

### ISSUE THREAD

**Issue by Pinkal7600** [fleaflet/flutter_map#725]
I am using the northEast and southWest LatLng to set bounds and it's working perfectly. But, as I compare to the leaflet by doing the same process. I get different bounds in output. How can I make them similar?

**flutter_map:**

    MapOptions mapOptions = MapOptions(
      center: LatLng(0, 0),
      onTap: _handleTap,
      minZoom: 1,
      maxZoom: 4,
      crs: CrsSimple(),
    );

    LatLng northEast = MapState(mapOptions)
        .unproject(CustomPoint(imageWidth, 0), mapOptions.maxZoom);
    LatLng southWest = MapState(mapOptions)
        .unproject(CustomPoint(0, imageHeight), mapOptions.maxZoom); 

    LatLngBounds latLngBonds = LatLngBounds(southWest, northEast);
    mapOptions.bounds = latLngBonds;

**leaflet:**
 
    this.previewMap = L.map('previewMap', {
      minZoom: 1,
      maxZoom: 4,
      center: [0, 0],
      crs: L.CRS.Simple,
    })

    const northEast = this.previewMap.unproject([width, 0], this.previewMap.getMaxZoom())
    const southWest = this.previewMap.unproject([0, height], this.previewMap.getMaxZoom())

    const bounds = new L.LatLngBounds(southWest, northEast)

**Output:**

**flutter_map**:
northEast Lat : -0.0, Lng : 0.5986328125
southWest Lat : -0.626953125, Lng : 0.0

**leaflet**:
northEast Lat : -0.0, Lng : 153.25
southWest Lat : -160.5, Lng : 0.0

I want to make the output bonds similar for flutter_map and leaflet

**Comment by ibrierley** [comment]
You can convert between them with project/unproject, you may have to look at /map/map.dart to see if you have access to those methods, or copy them and use if you don't maybe.

**Comment by Pinkal7600** [comment]
@ibrierley, I am not getting any idea about what things need to be changed to get a similar output of LatLngBounds. I checked project/unproject in map.dart but it will use the current CRS `pointToLatLng` method.  Can you please help me with which things I can look in detail and modify to get similar output?

**Comment by ibrierley** [comment]
Apologies, I hadn't looked through the code, and assumed leaflet was
converted to a different format, but probably not, so ignore my prev
reply....

Just out of interest, can you dump out the maxzooms/image width/heights,
just to prove what they are when the unproject happens...

Ian

**Comment by Pinkal7600** [comment]
If I remove the maxzoom, I get below output

**Comment by Pinkal7600** [comment]
If I remove max zoom, I get below output:

northEast Lat : -0.0, Lng : 0.0011692047119140625
southWest Lat : -0.001224517822265625, Lng : 0.0

We cannot dump `CustomPoint` which I use with image height and width.

**Comment by ibrierley** [comment]
Should be able to dump the custom points params (x,y or whatever).

I just think you need to prove the data you are entering into each is
exactly the same if you are getting different outputs first.

Ian

**Comment by Pinkal7600** [comment]
When I pass 0 as height, width, and max zoom for flutter_map and leaflet. I get the same output as below for `LatLngBounds`.

northwest :: lat : -0.0, lng : 0.0
southEast :: lat : -0.0, lng : 0.0

**Comment by ibrierley** [comment]
Think I'm lost on what you're trying to actually achieve here, why you are entering in 0 as width/height.

If it was me, main thing I would be testing for, with a different output, is if the inputs are exactly the same. What is the imageWidth/Height in the flutter_map version, what is the width/height in the leaflet version. Also just print the zoomlevels you use before calling the sub, just to be 100% sure all the variables are the same going in.

After that though, I'm not sure what the problem is you are actually having and if this is a redherring...what problem are you actually having ? Is it code afterwards which relies on those bounds that doesn't like those numbers ?

**Comment by Pinkal7600** [comment]
@ibrierley Apologies, I misunderstand the thing you want it. 

I used below height and width and the max zoom is already defined in the first post message which is 4 for both flutter_map and leaflet.

 double imageHeight = 2568;
 double imageWidth = 2452;

**Comment by Pinkal7600** [comment]
I found the solution. As compared to the leaflet we were missing something. So, I have made some changes as below and **also remove the `latlong: ^0.6.1` package and used `maps_toolkit: ^1.1.0+2` package for LatLng and some maths function for latlng as `latlong` packages is not maintained. But in `maps_toolkit` it also required modification in `maps_toolkit.dart` as we need to add `export 'src/math_util.dart';` to use math function for latlng.**

**Added below to override function in `CrsSimple` class.**

    @override
    num scale(double zoom) {
      return math.pow(2, zoom);
    }

    @override
    num zoom(double scale) {
      return math.log(scale) / math.ln2;
    }

**And modify the return of `unproject` of `_LonLat` class as below**

    @override
    LatLng unproject(CustomPoint point) {
      return LatLng(point.y, point.x);
    //    return LatLng(inclusiveLat(point.y), inclusiveLng(point.x)); // old return 
    }
